core/src/main/java/com/novoda/noplayer/internal/exoplayer/ExoPlayerFacade.java

```diff
@@ -139,8 +138,14 @@ private void setMovieAudioAttributes(SimpleExoPlayer exoPlayer) {
         }
     }
 
-    private void attachToSurface(SurfaceHolder surfaceHolder) {
-        exoPlayer.setVideoSurfaceHolder(surfaceHolder);
+    private void attachToSurface(PlayerSurfaceHolder playerSurfaceHolder) {
+        if (playerSurfaceHolder.containsSurfaceView()) {
+            exoPlayer.setVideoSurfaceView(playerSurfaceHolder.getSurfaceView());
+        } else if (playerSurfaceHolder.containsTextureView()) {
+            exoPlayer.setVideoTextureView(playerSurfaceHolder.getTextureView());
```

Comment:
do we know what the difference was between `set*View` and `setSurface` that was causing the glitchy behaviour in the previous solution?

Reply:
afaict internally the exoplayer holds a field for either a surface holder or a surface, I reckon that there's some different code path where a holder is used in lieu of a surface